tests/integration/test_renewals.cairo

```diff
@@ -0,0 +1,221 @@
+    tempvar renewal_contract;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        expect_revert(error_message="Renewer has not activated renewals for this domain")
+    %}
+    let (wont_renew) = Renewal.is_renewing(renewal_contract, TH0RGAL_STRING, 456);
+    assert wont_renew = 0;
+
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+
+    return ();
+}
+
+
+@external
+func test_renew_fail_expiry{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    buy_domain(60);
+    tempvar renewal_contract;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        expect_revert(error_message="Domain is not set to expire within a month")
+        stop_prank_callable = start_prank(456, target_contract_address=ids.renewal_contract)
+    %}
+
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (will_renew) = Renewal.is_renewing(renewal_contract, TH0RGAL_STRING, 456);
+    assert will_renew = 1;
+
+    %{ stop_prank_callable() %}
+
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+
+    return ();
+}
+
+@external
+func test_renew_domain{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    buy_domain(60);
+    tempvar renewal_contract;
+    tempvar naming;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        ids.naming = context.naming_contract
+        stop_prank_callable = start_prank(456, target_contract_address=ids.renewal_contract)
+        warp(5180000, context.renewal_contract)
+        stop_mock = mock_call(123, "transferFrom", [1])
+    %}
+
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (will_renew) = Renewal.is_renewing(renewal_contract, TH0RGAL_STRING, 456);
+    assert will_renew = 1;
+
+    %{ stop_prank_callable() %}
+
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+
+    assert_expiry(TH0RGAL_STRING, (60 * 86400) + (365 * 86400) + 1);
+
+    %{ stop_mock() %}
+
+    return ();
+}
+
+@external
+func test_renew_expired_domain{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    buy_domain(60);
+    tempvar renewal_contract;
+    tempvar naming;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        ids.naming = context.naming_contract
+        stop_prank_callable = start_prank(456, target_contract_address=ids.renewal_contract)
+        warp(10000000, context.renewal_contract)
+        stop_mock = mock_call(123, "transferFrom", [1])
+    %}
+
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (will_renew) = Renewal.is_renewing(renewal_contract, TH0RGAL_STRING, 456);
+    assert will_renew = 1;
+
+    %{ stop_prank_callable() %}
+
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+
+    assert_expiry(TH0RGAL_STRING, (60 * 86400) + (365 * 86400) + 1);
+
+    %{ stop_mock() %}
+
+    return ();
+}
+
+@external
+func test_renew_domains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
```

Comment:
Add comments on the different block of this function again for easy readability. 

See it a bit like `jest` test with js where we have `describe` and `it` loops. It permits us to decompose tests and it's easy to read. 

Would love to do something like this here. 

Examples here ==>  https://github.com/starknet-id/resolver_delegation/blob/7ec7f605d1646a281bd66c4cefa012721002d38a/tests/test_braavos.cairo#L24